epysurv/simulation/seasonal_noise.py

```diff
@@ -95,3 +97,100 @@ def simulate(
+    negative binomial model we draw case counts from.
+
+    Parameter
+    ----------
+    beta
+        trend parameter that controls the influence of the current week on :math:`\\mu`.
+    gamma_1
+        seasonality parameter to model the cos of the Fourier term.
+    gamma_2
+        seasonality parameter to model the sin of the Fourier term.
+    phi
+        dispersion parameter that regulates the overdispersion compared to the Poisson distribution
+        (:math:`\\phi * \\mu`)
+    seasonality_length
+        models the seasonality on a annual scope. If set to 0, no seasonality, if set to 1 then we model annual
+        seasonality, if set to 2 biannual seasonality and so forth.
+    theta
+        baseline frequency of cases.
+    seed
+        a seed for the random number generation.
+
+
+    References
+    ----------
+    An improved algorithm for outbreak detection in multiple surveillance system
+    https://doi.org/10.1002/sim.5595
+    """
+    beta: float = 0.003
+    gamma_1: float = 0.2
+    gamma_2: float = -0.4
+    phi: float = 1.0
+    seasonality_length: int = 1
+    seed: Optional[int] = None
+    theta: float = 1.5
+
+    def _seasonality(self, week: int):
+        """A Fourier-based seasonality term to model the season-depended case counts
+
+        Parameter
+        ----------
+        week
+            The week to model the season-based case count
+        """
+        return sum(
+            [
+                self.gamma_1 * np.cos((2 * np.pi * year * week) / 52)
+                + self.gamma_2 * np.sin((2 * np.pi * year * week) / 52)
+                for year in range(1, self.seasonality_length + 1)
+            ]
+        )
+
+    def simulate(self, length: int) -> pd.DataFrame:
+        """
+        Parameters
+        -------
+        length
+            number of weeks to model, default 100.
+
+        Returns
+        -------
+        A DataFrame of an endemic time series that contains n weeks where n=``length``.
+        Each row in the DataFrame's represents one time steps where each step is equivalent to one calender week.
+        It contains a ``mean`` column which is the mean case count according to the sinus based model.
+        And finally, it contains a column ``n_cases`` that consists of the generates case counts
+        based on the sinus model
+        """
+        if self.seed:
+            np.random.seed(self.seed)
+        mu_s = [
+            np.exp(self.theta + self.beta * week + self._seasonality(week))
+            for week in range(length)
+        ]
+        if self.phi == 1:
+            cases = [poisson.rvs(mu, size=1)[0] for mu in mu_s]
+        else:
+            cases = []
+            for mu in mu_s:
+                r = np.float(mu / (self.phi - 1))
+                p = r / (r + mu)
+                cases.append(nbinom.rvs(r, p, size=1)[0])
+        return pd.DataFrame({"n_cases": cases, "n_outbreak_cases": [0] * len(cases)})
```

Comment:
I set it to zero, as in the other simulation algorithms, because the detection algorithms require the `n_outbreak_cases ` column. So wouldn't it be wrong to leave out the column? Therefore, the only valid option would be to set it to `NaN` (?)

Reply:
I mean to change the detection algorithms so they do not require this any more. But it is a design decision that we should consider with care.

Reply:
Maybe it would be better to allow passing different variations of DataFrames to the detection algorithms and based whether it contains an `n_outbreak_cases ` column the algorithm could incorporate the information or it would leave it out. The latter would equal to assigning only NaNs to `n_oubtreak_cases` column. If the algorithm strictly requires the column an exception could be thrown.